File: src/exercicios/06_DepuracaoTestes/03-iniciante.c
/**      @file: 3-iniciante.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo de erro simples para depuração.
 *
 * Baseado no livro "The Practice of Programming", de Brian
 * W. Kernighan e Rob Pike (Addison-Wesley, 1999). */

#include <stdio.h>
#include <stdlib.h>


/* Especificação de algum comportamento que lida com
argumentos da linha de comando (ex: o comando gcc).

Exemplo de uso para depuração (supondo que este programa seja
o executável "3-iniciante"):

./3-iniciante -o meu_executavel -f123 -t500 1-iniciante.c

O resultado esperado é:

executavel [meu_executavel]
    inicio [123]
       fim [500]
*/

int main(int argc, char **argv) {
    int i, inicio, fim;
    char *executavel = NULL;

    /* Avaliação das opções dadas na linha de comando. */
    for(i=1; i < argc; i++) {
        if(argv[i][0] != '-') /* opções terminadas */
            break;
        switch(argv[i][1]) {
        case 'o':             /* nome do executável */
            executavel = argv[i];
            break;
        case 'f':             /* valor inicial */
            inicio = atoi(argv[i]);
            break;
        case 't':             /* valor final */
            fim = atoi(argv[i]) ;
            break;
        default:
            break;
        }
    }
                                             /* Resultado esperado: */
    printf("executavel [%s].\n", executavel);/* executavel [meu_executavel] */
    printf("    inicio [%d].\n", inicio);    /*     inicio [123] */
    printf("       fim [%d].\n", fim);       /*        fim [500] */

    return 0;
}

File: src/exercicios/06_DepuracaoTestes/03-iniciante/03-iniciante.py
#  -*- coding: utf-8 -*-
#    @package: 03-iniciante.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo de erro simples para depuração.


# Especificação de algum comportamento que lida com
# argumentos da linha de comando (ex: o comando gcc).
#
# Exemplo de uso para depuração:
#
# python 03-iniciante.py -o meu_executavel -f123 -t500 01-iniciante.py
#
# O resultado esperado é:
#
# executavel [meu_executavel]
#     inicio [123]
#        fim [500]


import sys


if __name__ == '__main__':
    inicio = fim = executavel = None

    # Avaliação das opções dadas na linha de comando.
    for i in range(1, len(sys.argv)):
        if sys.argv[i][0] != '-':       #  opções terminadas
            continue
        if sys.argv[i][1] == 'o':       #  nome do executável
            executavel = sys.argv[i]
        elif sys.argv[i][1] == 'f':       #  valor inicial
            inicio = int(sys.argv[i])
        elif sys.argv[i][1] == 't':       #  valor final
            fim = int(sys.argv[i])
                                                  # Resultado esperado:
    print('executavel [{}].'.format(executavel))  # executavel [meu_executavel]
    print('    inicio [{}].'.format(inicio))      #     inicio [123]
    print('       fim [{}].'.format(fim))         #        fim [500]
